generative_model_training/05_plot_labels_sampling_performance.py

```diff
@@ -0,0 +1,511 @@
+        ymax="auroc_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ### Test Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="auroc_mean",
+        ymin="auroc_lower_ci",
+        ymax="auroc_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ## Binary Cross Entropy Loss
+
+# ### Tune Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='tune'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="bce_mean",
+        ymin="bce_lower_ci",
+        ymax="bce_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ### Test Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="bce_mean",
+        ymin="bce_lower_ci",
+        ymax="bce_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# # Full Text
+
+# ## Area Under Precision Recall Curve (AUPR)
+
+# ### Tune Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='full_text'")
+        >> ply.query("model=='tune'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="aupr_mean",
+        ymin="aupr_lower_ci",
+        ymax="aupr_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ### Test Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='full_text'")
+        >> ply.query("model=='test'")
```

Comment:
Applicable to all these plots, but is there a baseline to compare these performances against? Or would it require some hard graph permutation problem to calculate?

Reply:
>Applicable to all these plots, but is there a baseline to compare these performances against? Or would it require some hard graph permutation problem to calculate?

I forgot to mention this point, but all other points are being compared against the 0th point. The 0 point is distant supervision (labeling sentences based on databases alone) and the goal is to see how performance improves or decreases from that. There might not be a zero point for full text and I'll double check as I make changes to this notebook.

Reply:
Ah I forgot to run the baseline for full text. Good catch will need to run and upload!